packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -78,6 +92,9 @@ export class Snapshot {
   }
 
   public get targetId() {
+    if (this.model.targetId === undefined) {
```

Comment:
What are the scenario where this ID would be undefined ?

Reply:
Multiple parameters from the snapshotModel interface are optional as opposed to what is described by the Migration API. 
I'm checking with the backend team if a pull request on the platform-client is needed.

Reply:
https://github.com/coveo/platform-client/pull/339